internal/credentials/spiffe_test.go

```diff
@@ -178,3 +181,25 @@ func (s) TestSPIFFEIDFromState(t *testing.T) {
 		})
 	}
 }
+
+func (s) TestSPIFFEIDFromCert(t *testing.T) {
+	data, err := ioutil.ReadFile(testdata.Path("x509/spiffe_cert.pem"))
+	if err != nil {
+		t.Fatalf("Failed to load the test credentials: %v", err)
+	}
+	block, rest := pem.Decode(data)
+	if len(rest) != 0 {
```

Comment:
I think it will be better to check `block != nil` and ignore `rest` with an `_`.